### PR TITLE
Timeline feature does not respect security constraint set by "add-group-repo" kie-cli command

### PR DESCRIPTION
**Pull request by ederign**
Depends on https://github.com/droolsjbpm/kie-uberfire-extensions/pull/9.

### CODE REVIEW

**Comment by ederign** [comment]
fixing.

**Comment by manstis** [comment]
Licenses!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventOUConstraint.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package org.kie.workbench.common.screens.social.hp.security;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventOUConstraint.java]: License!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package org.kie.workbench.common.screens.social.hp.security;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: License!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/security/SocialEventOUConstraintTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
+package org.kie.workbench.common.screens.social.hp.security;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/security/SocialEventOUConstraintTest.java]: License!

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraintTest.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
+package org.kie.workbench.common.screens.social.hp.security;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/test/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraintTest.java]: License!

**Comment by ederign** [comment]
@csadilek  ready to merge!

**Comment by manstis** [comment]
+1 to merge.

**Comment by ederign** [comment]
checking....

**Comment by ederign** [comment]
@psiroky  "Problem copying files : No space left on device"

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+
+@ApplicationScoped
+public class SocialEventRepositoryConstraint implements SocialSecurityConstraint {
+
+    @Inject
+    OrganizationalUnitService organizationalUnitService;
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: private?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Inject
+    AuthorizationManager authorizationManager;
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: private?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Inject
+    RepositoryServiceImpl repositoryService;
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: private?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Inject
+    User identity;
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: private?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+    Set<Repository> authorizedRepos = new HashSet<Repository>();
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: private?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        final FileSystem fileSystem = path.getFileSystem();
+        return repositoryService.getRepository( fileSystem );
+    }
+
+    private boolean isAProjectEvent( SocialActivitiesEvent event ) {
+        return event.getLinkType().equals( SocialActivitiesEvent.LINK_TYPE.CUSTOM )
+                && event.getType().equals( ProjectEventType.NEW_PROJECT.name() );
+    }
+
+    Set<Repository> getAuthorizedRepositories() {
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-backend/src/main/java/org/kie/workbench/common/screens/social/hp/security/SocialEventRepositoryConstraint.java]: private?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/java/org/kie/workbench/common/screens/social/hp/client/homepage/DefaultSocialLinkCommandGenerator.java [hunk 11]
```diff
@@ -35,14 +30,19 @@
...
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+import java.util.Set;
+
 @ApplicationScoped
 public class DefaultSocialLinkCommandGenerator {
 
     @Inject
-    private PlaceManager placeManager;
+    PlaceManager placeManager;
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/java/org/kie/workbench/common/screens/social/hp/client/homepage/DefaultSocialLinkCommandGenerator.java]: private + @Inject @ contructor?

File: kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/java/org/kie/workbench/common/screens/social/hp/client/homepage/DefaultSocialLinkCommandGenerator.java [hunk 12]
```diff
@@ -35,14 +30,19 @@
...
 
     @Inject
-    private Event<SocialFileSelectedEvent> socialFileSelectedEvent;
+    Event<SocialFileSelectedEvent> socialFileSelectedEvent;
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-social-home-page/kie-wb-common-social-home-page-client/src/main/java/org/kie/workbench/common/screens/social/hp/client/homepage/DefaultSocialLinkCommandGenerator.java]: private + @Inject @ contructor?

**Comment by ederign** [comment]
Updated: https://github.com/droolsjbpm/kie-wb-common/pull/139

**Pull request closed without merging**